sql/expression/in.go

```diff
@@ -62,9 +59,15 @@ func (in *InTuple) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
 	}
 
 	if left == nil {
-		return nil, err
+		return nil, nil
 	}
 
+	// The NULL handling for IN expressions is tricky. According to
```

Comment:
This threw me off at first when I was looking at the tests earlier in the PR, and had to look it up. This stuff is not straightforward lol